src/pages/memos/[functionId]/MemosForm.js

```diff
@@ -0,0 +1,531 @@
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import CustomDatePicker from 'src/components/Inputs/CustomDatePicker'
+import { formatDateFromApi } from 'src/lib/date-helper'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import CustomTextArea from 'src/components/Inputs/CustomTextArea'
+import { FinancialRepository } from 'src/repositories/FinancialRepository'
+import { SystemFunction } from 'src/resources/SystemFunction'
+import { useDocumentType } from 'src/hooks/documentReferenceBehaviors'
+
+export default function MemosForm({ labels, access, recordId, functionId, getEndpoint }) {
+  const { documentType, maxAccess, changeDT } = useDocumentType({
+    functionId: functionId,
+    access: access,
+    enabled: !recordId
+  })
+
+  const [initialVatPct, setInitialVatPct] = useState('')
+
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: FinancialRepository.FiMemo.qry
+  })
+
+  // const getSetEndpoint = functionId => {
+  //   switch (functionId) {
+  //     case SystemFunction.CreditNote:
+  //       return FinancialRepository.CreditNote.set
+  //     case SystemFunction.DebitNote:
+  //       return FinancialRepository.DebitNote.set
+  //     case SystemFunction.ServiceBill:
+  //       return FinancialRepository.ServiceBillReceived.set
+  //     case SystemFunction.ServiceInvoice:
+  //       return FinancialRepository.ServiceInvoice.set
+  //     default:
+  //       return null
+  //   }
+  // }
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: recordId || null,
+      dtId: documentType?.dtId || null,
+      reference: '',
+      date: new Date(),
+      plantId: '',
+      currencyId: '',
+      status: '',
+      accountId: '',
+      amount: '',
+      baseAmount: '',
+      functionId: functionId,
+      vatPct: '',
+      exRate: '',
+      rateCalcMethod: '',
+      subtotal: '',
+      notes: '',
+      vatAmount: '',
+      isSubjectToVAT: false
+    },
+    maxAccess,
+    enableReinitialize: false,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      amount: yup.number().required(' '),
+      currencyId: yup.string().required(' '),
+      accountId: yup.string().required(' '),
+      subtotal: yup.number().required(' '),
+      date: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      const recordId = obj.recordId
```

Comment:
can we remove this line and use obj.recordId directly below?

Reply:
there is an if statement based on record id